starter.yaml rewrite (#1154)
kyma-project/test-infra: development/tools/cmd/yamlmerge/README.md

> # YAML merge

Reviewer: Why do we need to merge the YAML files? Can't we just run `kubectl apply -f` in the components folder to deploy all the resources to the cluster?

Author: We can, k8s has a setup like this and they use bazel to merge it into starter.yaml. The problem would be just, that if we move all the `components` into the main folder, the amount of files in that folder would just grow and it would be hard to keep an overview. If we have the merger tool, we can work on the separate files, merge them into starter.yaml at the end and do `kubectl apply -f` to the main folder (assuming it does not include yaml files from subfolders)

## Overview

> This command-line tool enables merging YAML files on a given path into one file. It simply creates a`starter.yaml` file from the component configurations under the `/prow/cluster/components` path, and generates a file that includes the content of `/prow/cluster/starter.yaml`.

Reviewer 2: ```suggestion
This command line tool enables merging yaml files into one single file. For the operation to work, the yaml files must follow the path with a given extension. For example, the tool creates a`starter.yaml` file from the component configurations residing under the `/prow/cluster/components` path, and puts the file under  `/prow/cluster/starter.yaml`. Then it generates a file out of the content of the `starter.yaml` file. 
```
Question: is it always the /prow/cluster/components path? I assume it can be a different path - just all the yaml files must have the same, so I would document this specific one as an example. Also, what kind of file is the target file? is it also a yaml file?

## Usage

For security reasons, the dry-run mode is the default one.
To run it, use:
```bash
go run main.go -path /path/to/components/folder -target /path/to/target/file
```

To turn the dry-run mode off, use:
```bash
go run main.go -path /path/to/components/folder -target /path/to/target/file -dryRun=false
```

### Flags

See the list of available flags:

| Name                      | Required | Description                                                                                          |
| :------------------------ | :------: | :--------------------------------------------------------------------------------------------------- |
| **--path**                |   Yes    | Path to the folder containing the yaml files to merge.

> | **--target**              |   Yes    | Path of the file, that is target to collected files' merged content.

Reviewer 2: ```suggestion
| **--target**              |   Yes    | Path of the file which includes the content of the merged files.
```

# YAML merge

## Overview

> This command line tool enables merging yaml files into one single file. For the operation to work, the yaml files must follow the path with a given extension. For example, the tool creates a `starter.yaml` file from the component configurations residing under the `/prow/cluster/components` path, and puts the file under  `/prow/cluster/starter.yaml` if called like so:

Reviewer 2: ```suggestion
This command line tool enables merging yaml files into one single file. For the operation to work, the yaml files must follow the same source path. 
For example, the following command allows the tool to merge the component configurations residing under the `/prow/cluster/components` path and place the resulting target file under `/prow/cluster/starter.yaml`. 
```

```bash
go run main.go -path /prow/cluster/components -target /prow/cluster/starter.yaml
```

> If `starter.yaml` in the given path does not exist it will be created.

Reviewer 2: ```suggestion
>**NOTE**:  If the `starter.yaml` file does not already exist under a given path, it will be created.
```
